### PR TITLE
Rename GPU thread to raster thread in API docs

### PR DESCRIPTION
**Pull request by filiph**
## Description

As per https://github.com/flutter/flutter/issues/29443, we are renaming the GPU thread to “raster thread”. The main reason is that most developers assume the GPU thread is running on the GPU, which leads them astray.

DevTools, the engine, and the website are already updated. This PR will complete the necessary code changes. See https://github.com/flutter/flutter/issues/29443#issuecomment-602791835 for the whole picture.

I purposefully didn’t touch the tracing event name constant `GPURasterizer::Draw`. Not only does it already convey what it does well even after the GPU->Raster rename, but changing the constant would break folks who depend on that name in their perf benchmarks.

This is purely a documentation change.

## Related Issues

* https://github.com/flutter/flutter/issues/29443#issuecomment-602791835

## Tests

This is purely a documentation change. I believe no tests are needed for it.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: packages/flutter/lib/src/painting/shader_warm_up.dart [hunk 1]
```diff
@@ -24,7 +24,7 @@ import 'package:flutter/foundation.dart';
 /// done before calling [runApp].
 ///
 /// To determine whether a draw operation is useful for warming up shaders,
-/// check whether it improves the slowest GPU frame. Also, tracing with
+/// check whether it improves the slowest raster frame. Also, tracing with
```
Comment (liyuqian) [inline packages/flutter/lib/src/painting/shader_warm_up.dart]: Nit: `slowest frame rasterization time` might be better than `slowest raster frame`.

File: packages/flutter/lib/src/painting/shader_warm_up.dart [hunk 2]
```diff
@@ -72,7 +72,7 @@ abstract class ShaderWarmUp {
   /// `flutter screenshot --observatory-uri=<uri> --type=skia`
   /// and analyze it with https://debugger.skia.org.
   /// Alternatively, one may run the app with `flutter run --trace-skia` and
-  /// then examine the GPU thread in the observatory timeline to see which
+  /// then examine the raster thread in the observatory timeline to see which
```
Comment (liyuqian) [inline packages/flutter/lib/src/painting/shader_warm_up.dart]: Nit: formerly known as the GPU thread

File: packages/flutter/lib/src/painting/binding.dart [hunk 3]
```diff
@@ -44,9 +44,9 @@ mixin PaintingBinding on BindingBase, ServicesBinding {
   /// installation or a data wipe. The warm up does not block the main thread
   /// so there should be no "Application Not Responding" warning.
   ///
-  /// Currently the warm-up happens synchronously on the GPU thread which means
-  /// the rendering of the first frame on the GPU thread will be postponed until
-  /// the warm-up is finished.
+  /// Currently the warm-up happens synchronously on the raster thread which
+  /// means the rendering of the first frame on the raster thread will be
+  /// postponed until the warm-up is finished.
```
Comment (liyuqian) [inline packages/flutter/lib/src/painting/binding.dart]: Nit: formerly known as the GPU thread

File: dev/integration_tests/ui/lib/screenshot.dart [hunk 4]
```diff
@@ -64,7 +64,7 @@ List<Widget> _buildRows(int count) {
 }
 
 /// Builds cells that are known to take time to render causing a delay on the
-/// GPU thread.
+/// raster thread.
```
Comment (liyuqian) [inline dev/integration_tests/ui/lib/screenshot.dart]: Nit: formerly known as the GPU thread

File: dev/integration_tests/ui/lib/screenshot.dart [hunk 5]
```diff
@@ -6,8 +6,8 @@ import 'package:flutter/material.dart';
 import 'package:flutter_driver/driver_extension.dart';
 
 /// This sample application creates a hard to render frame, causing the
-/// driver script to race the GPU thread. If the driver script wins the
-/// race, it will screenshot the previous frame. If the GPU thread wins
+/// driver script to race the raster thread. If the driver script wins the
+/// race, it will screenshot the previous frame. If the raster thread wins
```
Comment (liyuqian) [inline dev/integration_tests/ui/lib/screenshot.dart]: Nit: formerly known as the GPU thread

File: dev/benchmarks/complex_layout/test_memory/scroll_perf.dart [hunk 6]
```diff
@@ -34,9 +34,9 @@ Future<void> main() async {
   ));
   await SchedulerBinding.instance.endOfFrame;
 
-  /// Wait 50ms to allow the GPU thread to actually put up the frame. (The
-  /// endOfFrame future ends when we send the data to the engine, before the GPU
-  /// thread has had a chance to rasterize, etc.)
+  /// Wait 50ms to allow the raster thread to actually put up the frame. (The
+  /// endOfFrame future ends when we send the data to the engine, before
+  /// the raster thread has had a chance to rasterize, etc.)
```
Comment (liyuqian) [inline dev/benchmarks/complex_layout/test_memory/scroll_perf.dart]: Nit: formerly known as the GPU thread

File: packages/flutter_driver/lib/src/driver/driver.dart [hunk 7]
```diff
@@ -532,27 +533,27 @@ abstract class FlutterDriver {
   ///        `screenshot()`. The gap is random because it is determined by the
   ///        unpredictable network communication between the driver process and
   ///        the application. If this gap is too short, which it typically will
-  ///        be, the screenshot is taken before the GPU thread is done
+  ///        be, the screenshot is taken before the raster thread is done
```
Comment (liyuqian) [inline packages/flutter_driver/lib/src/driver/driver.dart]: Nit: formerly known as the GPU thread

File: packages/flutter_driver/lib/src/driver/driver.dart [hunk 8]
```diff
@@ -532,27 +533,27 @@ abstract class FlutterDriver {
...
   ///        With this delay, if we're lucky:
   ///        -------------------------------------------------------------------
   ///        UI    : <-- build -->
-  ///        GPU   :               <-- rasterize -->
+  ///        Raster:               <-- rasterize -->
   ///        Gap   :              |    2 seconds or more   |
   ///        Driver:                                        <-- screenshot -->
   ///
-  ///        The two-second gap should be long enough for the GPU thread to
+  ///        The two-second gap should be long enough for the raster thread to
```
Comment (liyuqian) [inline packages/flutter_driver/lib/src/driver/driver.dart]: Nit: formerly known as the GPU thread

File: packages/flutter_driver/lib/src/driver/timeline_summary.dart [hunk 9]
```diff
@@ -74,8 +74,8 @@ class TimelineSummary {
     return _percentileInMillis(_extractGpuRasterizerDrawDurations(), p);
   }
 
-  /// The number of frames that missed the [kBuildBudget] on the GPU and
-  /// therefore are in the danger of missing frames.
+  /// The number of frames that missed the [kBuildBudget] on the raster thread
+  /// and therefore are in the danger of missing frames.
```
Comment (liyuqian) [inline packages/flutter_driver/lib/src/driver/timeline_summary.dart]: Nit: formerly known as the GPU thread

**Review by liyuqian — APPROVED** [review]
LGTM with a few nits :)

File: packages/flutter/lib/src/painting/shader_warm_up.dart (hunk 1 above)
Comment (filiph) [reply packages/flutter/lib/src/painting/shader_warm_up.dart]: Ack, will fix.

File: packages/flutter/lib/src/painting/shader_warm_up.dart (hunk 2 above)
Comment (filiph) [reply packages/flutter/lib/src/painting/shader_warm_up.dart]: I didn't add the "formerly known" parts in the deepest depths of the framework, and instead put them only in the files most likely seen by a beginner/intermediate user.

I'm not set on that decision, though. What do you think?

File: packages/flutter_driver/lib/src/driver/driver.dart (hunk 7 above)
Comment (filiph) [reply packages/flutter_driver/lib/src/driver/driver.dart]: This is already explained a few lines above. I think it's better to put the "formerly known" explanation only once per each doc comment.

File: packages/flutter/lib/src/painting/shader_warm_up.dart (hunk 2 above)
Comment (liyuqian) [reply packages/flutter/lib/src/painting/shader_warm_up.dart]: Sounds good to me. Let's omit "formerly known" in those places.

**Comment by zanderso** [comment]
@filiph what is the status of this PR?

**Comment by filiph** [comment]
Ah, thanks for the nudge! I just implemented the change suggested by @liyuqian. Now, he and I are happy with it. 

It's a docs-only change. Should we wait for someone on the framework team to approve as well? (I requested @dnfield since he was in the #29443 discussion.)

**Review by zanderso — APPROVED** [review]
This seems safe to land.

**Review by goderbauer — APPROVED** [review]
LGTM

**Pull request merged**